mgmt/mcumgr: Commits that internalize cbor_nb_* functions and remove zephyr/mgmt/mcumgr/buf.h

de-nordic opened the pull request:
Three commits:
 1) internalizes cbor_nb_* functions as there is no reason to export them;
 2) Moves rest of the zephyr/mgmt/mcumgr/buf.h into subsys/mgmt/mcumgr/lib/smp/include/smp/smp.h and removes the former;
 3) Moves mcumgr_buf_alloc and mcumgr_buf_free out of buf.c to smp.c, renames them smp_packet_alloc and smp_packet_free, and remove the buf.c file completely.

de-nordic committed:
mgmt/mcumgr: Move and internalize cbor_nb_*_init functions

The commit moves functions used for initialization of CBOR encoding
and decoding to the only unit that is supposed to use them.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
mgmt/mcumgr: Remove mgmt/mcumgr/buf.h

SMP buffer allocation functions have been moved to smp/smp.h,
and buf.h has been removed.
Definitions of cbor_nb_reader and cbor_nb_writer have also been moved.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm commented on subsys/mgmt/mcumgr/transport/smp_bt.c:
```diff
@@ -17,10 +17,11 @@
-#include <zephyr/mgmt/mcumgr/buf.h>
 
 #include <zephyr/mgmt/mcumgr/smp.h>
+#include <zcbor_common.h>
```
No zcbor usage in this file

de-nordic replied:
Hm. It really baffles me how come I did add the <zcbor_common.h> to so many files, where it is not needed.

nordicjm commented on subsys/mgmt/mcumgr/smp_reassembly.c:
```diff
@@ -7,8 +7,8 @@
 #include <zephyr/net/buf.h>
-#include <zephyr/mgmt/mcumgr/buf.h>
 #include <zephyr/mgmt/mcumgr/smp.h>
+#include <zcbor_common.h>
```
Not using zcbor functions in this file

nordicjm commented on subsys/mgmt/mcumgr/lib/mgmt/src/mgmt.c:
```diff
@@ -4,13 +4,14 @@
-#include <zephyr/mgmt/mcumgr/buf.h>
-#include <zcbor_encode.h>
 #include <zcbor_common.h>
+#include <zcbor_encode.h>
```
No zcbor functions being used in this file

nordicjm commented on subsys/mgmt/mcumgr/lib/smp/include/smp/smp.h:
```diff
@@ -25,14 +25,39 @@
 #ifndef H_SMP_
 #define H_SMP_
 
-#include "mgmt/mgmt.h"
```
This needs a zcbor and zephyr kernel include

de-nordic replied:
No it does not. Nothing in this header defines anything that is `struct net_buf` only `struct net_buf*`.
Definition of `struct net_buf` is only needed on implementation side.

nordicjm replied:
This places the onus of file includes on the person that includes this file, the end user should not have to add file includes to their files to be able to include this file without build errors

nordicjm commented on subsys/mgmt/mcumgr/smp.c:
```diff
@@ -8,11 +8,11 @@
-#include <zephyr/mgmt/mcumgr/buf.h>
-#include "mgmt/mgmt.h"
-#include "smp/smp.h"
+#include <zcbor_common.h>
```
No zcbor usage in this file

nordicjm replied:
Not sure why I marked this as resolved.

nordicjm commented on subsys/mgmt/mcumgr/transport/serial_util.c:
```diff
@@ -13,8 +13,10 @@
 #include <zephyr/sys/base64.h>
-#include <zephyr/mgmt/mcumgr/buf.h>
 #include <zephyr/mgmt/mcumgr/serial.h>
+#include <zcbor_common.h>
```
No zcbor function usage in this file

nordicjm commented on subsys/mgmt/mcumgr/transport/smp_shell.c:
```diff
@@ -12,15 +12,16 @@
+#include <zephyr/shell/shell_uart.h>
+#include <zephyr/drivers/uart.h>
+#include <syscalls/uart.h>
+#include <zcbor_common.h>
```
No zcbor usage in this file

nordicjm commented on subsys/mgmt/mcumgr/transport/smp_uart.c:
```diff
@@ -12,9 +12,10 @@
 #include <zephyr/net/buf.h>
-#include <zephyr/mgmt/mcumgr/buf.h>
 #include <zephyr/drivers/console/uart_mcumgr.h>
+#include <zcbor_common.h>
```
No zcbor usage in this file

nordicjm commented on subsys/mgmt/mcumgr/transport/smp_udp.c:
```diff
@@ -18,9 +18,10 @@
 #include <zephyr/net/socket.h>
 #endif
 #include <errno.h>
+#include <zcbor_common.h>
```
No zcbor usage in this file

de-nordic committed:
mgmt/mcumgr: Remove buf.c

buf.c have been providing net_buf pool allocator and dealocator
for SMP packets: mcumgr_buf_alloc and mcumgr_buf_free.
The functions have been moved to smp.c and renamed
smp_packet_alloc and smp_packet_free, respectively.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic commented on subsys/mgmt/mcumgr/lib/smp/include/smp/smp.h:
```diff
@@ -25,14 +25,41 @@
-struct mgmt_hdr;
 
 /** @typedef smp_tx_rsp_fn
  * @brief Transmits an SMP response packet.
```
`mgmt_streamer ` no longer exists.

nordicjm commented on subsys/mgmt/mcumgr/lib/smp/include/smp/smp.h:
```diff
@@ -25,14 +25,42 @@
 
-#include "mgmt/mgmt.h"
+#include <zephyr/net/buf.h>
+#include <zcbor_common.h>
```
Needs `zephyr/mgmt/mcumgr/smp.h` include for zephyr_smp_transport

nordicjm reviewed (commented):
`subsys/mgmt/mcumgr/smp.c` has a zcbor include with no zcbor usage in the file, it's been marked as resolved, but the include is still there
